pkgs/applications/networking/instant-messengers/dino/default.nix

```diff
@@ -40,37 +38,28 @@ stdenv.mkDerivation rec {
   buildInputs = [
     qrencode
     gobject-introspection
-    glib-networking
     glib
+    glib-networking # required for TLS support
+    libadwaita
     libgee
-    gnome.adwaita-icon-theme
     sqlite
     gdk-pixbuf
-    gtk3
+    gtk4
     libnotify
     gpgme
     libgcrypt
     libsoup
-    pcre
-    libepoxy
-    at-spi2-core
-    dbus
+    pcre2
     icu
     libsignal-protocol-c
-    librsvg
```

Comment:
How did you determine which dependencies to remove? For example, [this was needed for the about window](https://github.com/NixOS/nixpkgs/commit/091b710c3f4f5138afcec281f1a21a203366820c), but wasn't obvious — it didn't fail the build or anything.

Reply:
In general, I don't think we should be removing dependencies unless we can either [explain why they were added](https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence), and why that reasoning no longer applies, or why they're obviously not required and never have been.  Otherwise we risk runtime regressions.

Reply:
I build the update and tested things. Since upstream switched to gtk4 there are now unused packages which I wanted to remove.

Reply:
![image](https://user-images.githubusercontent.com/7258858/217403562-5e39563b-f83d-4b0a-8385-b1a95a253238.png)


The about screen now works without this dependency and I also couldn't find any reference of it in the source code anymore.